spec/services/usps_in_person_proofer_spec.rb

```diff
@@ -0,0 +1,203 @@
+        first_name: Faker::Name.first_name,
+        last_name: Faker::Name.last_name,
+        email: Faker::Internet.safe_email,
+        unique_id: '123456789',
+      )
+
+      subject.retrieve_token!
+      enrollment = subject.request_enroll(applicant)
+      expect(enrollment['enrollmentCode']).to be_present
+      expect(enrollment['responseMessage']).to be_present
+    end
+  end
+
+  describe '#request_proofing_results' do
+    it 'returns failed enrollment information' do
+      stub_request_token
+      stub_request_failed_proofing_results
+
+      applicant = double('applicant',
+        unique_id: '123456789',
+        enrollment_code: '123456789',
+      )
+
+      subject.retrieve_token!
+      proofing_results = subject.request_proofing_results(
+        applicant.unique_id,
+        applicant.enrollment_code,
+      )
+      expect(proofing_results['status']).to eq 'In-person failed'
+      expect(proofing_results['fraudSuspected']).to eq false
+    end
+
+    it 'returns passed enrollment information' do
+      stub_request_token
+      stub_request_passed_proofing_results
+
+      applicant = double('applicant',
+        unique_id: '123456789',
+        enrollment_code: '123456789',
+      )
+
+      subject.retrieve_token!
+      proofing_results = subject.request_proofing_results(
+        applicant.unique_id,
+        applicant.enrollment_code,
+      )
+      expect(proofing_results['status']).to eq 'In-person passed'
+      expect(proofing_results['fraudSuspected']).to eq false
+    end
+
+    it 'returns in-progress enrollment information' do
+      stub_request_token
+      stub_request_in_progress_proofing_results
+
+      applicant = double('applicant',
+        unique_id: '123456789',
+        enrollment_code: '123456789',
+      )
+
+      subject.retrieve_token!
+      proofing_results = subject.request_proofing_results(
+        applicant.unique_id,
+        applicant.enrollment_code,
+      )
+      expect(proofing_results['responseMessage']).to eq(
+        'Customer has not been to a post office to complete IPP',
+      )
+    end
+  end
+
+  describe '#request_enrollment_code' do
+    it 'returns enrollment information' do
+      stub_request_token
+      stub_request_enrollment_code
+      applicant = double('applicant',
+        unique_id: '123456789',
+      )
+
+      subject.retrieve_token!
+      enrollment = subject.request_enrollment_code(applicant)
+      expect(enrollment['enrollmentCode']).to be_present
+      expect(enrollment['responseMessage']).to be_present
+    end
+  end
+
+  def stub_request_token
+    stub_request(:post, %r{/oauth/authenticate}).to_return(
+      status: 200, body: UspsIppFixtures.request_token_response, headers: {},
+    )
+  end
+
+  def stub_request_facilities
+    stub_request(:post, %r{/ivs-ippaas-api/IPPRest/resources/rest/getIppFacilityList}).to_return(
+      status: 200, body: UspsIppFixtures.request_facilities_response, headers: {},
```

Comment:
Let's just drop `headers: {}` if it's blank for all these?

Reply:
👍🏼